mrmustard/math/tensor_networks/tensors.py

```diff
@@ -0,0 +1,328 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Classes for constructing tensors."""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from typing import List, Optional
+
+import numpy as np
+import uuid
+
+
+__all__ = ["Wire", "Tensor"]
+
+
+def random_int() -> int:
+    r"""
+    Returns a random integer obtained from a UUID
+    """
+    return uuid.uuid1().int
```

Comment:
maybe uuid4 is better, but yes this is better than what I had done 😅

Reply:
About this.. If I use `uuid4` or `np.random.randint` or similar functions, the TN tests fail. If I use `uuid1`, they pass. I wonder if this only happens on my machine.. :face_in_clouds:

Reply:
oh, that's interesting 🧐